src/mercurial/depository.ts

```diff
@@ -138,8 +137,50 @@ export class MercurialVaultDepository {
       vaultPda,
       mercurialVaultLpMint,
       depositoryLpTokenVault,
-      mercurialVaultCollateralTokenSafe
+      mercurialVaultCollateralTokenSafe,
+      mercurialVaultProgram
+    );
+  }
+
+  public async calculateProfitsValue(
```

Comment:
to speed up the merge of this branch, we could put this profit calculation function into another future incremental PR (we wouldnt need types/utils new files for this IDL change)

Reply:
it's been using in the test on uxd-program already for mercurial profit collection, let's include that in this PR